File: Proyecto.Ecommerce/Proyecto.Ecommerce.Aplicacion/Dtos/OrdenCarroDto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Proyecto.Ecommerce.Aplicacion.Dtos
{
    public class OrdenCarroDto
    {
        public List<OrdenItemsDto> OrdenItemsDtos { get; set; }
        public ClienteDto Cliente { get; set; }
        public MetodoEntregaDto MetodoEntrega { get; set; }
        public float SubTotal { get; set; }
        public int Iva { get; set; }
        public float Total { get; set; }
    }
}

File: Proyecto.Ecommerce.Front/src/app/models/OrdenCarroDto.ts
import { ClienteDto } from "./ClienteDto"
import { MetodoEntregaDto } from "./MetodoEntregaDto"
import { OrdenItemsDto } from "./OrdenItemsDto"

export interface OrdenCarroDto{
    ordenItemsDtos:OrdenItemsDto[],
    cliente:ClienteDto,
    metodoEntrega:MetodoEntregaDto,
    subTotal:number,
    iva:number,
    total:number,
}